lab_2_keywords_cooccurrence/main.py

```diff
@@ -120,10 +150,22 @@ def calculate_cumulative_score_for_candidates(candidate_keyword_phrases: KeyPhra
 
     In case of corrupt input arguments, None is returned
     """
-    if not type_check(candidate_keyword_phrases, list) or not type_check(word_scores, dict) or \
-            not all(token in word_scores for token in list(chain.from_iterable(candidate_keyword_phrases))):
+    if not is_not_empty_or_bad(candidate_keyword_phrases, list, tuple) or not is_not_empty_or_bad(word_scores, dict):
         return None
-    return {phrase: sum(word_scores[token] for token in phrase) for phrase in candidate_keyword_phrases}
+    for tuple_phrase in candidate_keyword_phrases:
+        for word in tuple_phrase:
+            if word not in word_scores.keys():
+                return None
+    keyword_phrases_with_scores = dict.fromkeys(candidate_keyword_phrases)
```

Comment:
why use `fromkeys`?

Reply:
it was easier to see for me how the dict fills up, but it crashed a type check ,__,